src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,68 @@
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  onFilterChange: React.Dispatch<React.SetStateAction<string>>;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  query: string;
+  selectFilter: string;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  onFilterChange,
+  setQuery,
+  query,
+  selectFilter,
+}) => {
+  const filterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    onFilterChange(event.target.value);
+  };
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Consider renaming the `onChange` function to `handleQueryChange` for better clarity and consistency with the `filterChange` function.

Reply:
> Consider renaming the `onChange` function to `handleQueryChange` for better clarity and consistency with the `filterChange` function.

Good point